src/main/cs/Infinispan/HotRod/Impl/RemoteCacheSWIGImpl.cs

```diff
@@ -253,6 +261,75 @@ public byte[] Execute(string scriptName, IDictionary<string, string> dict)
             return ret;
         }
 
+        public Task<V> GetAsync(K key)
+        {
+            return Task.Run(() => Get(key));
+        }
+
+        public Task<V> PutAsync(K key, V val, ulong lifespan = 0, TimeUnit lifespanUnit = TimeUnit.SECONDS, ulong maxIdleTime = 0, TimeUnit maxIdleUnit = TimeUnit.SECONDS)
+        {
+            if (!manager.IsStarted())
+            {
+                String message = "Cannot perform operations on a cache associated with an unstarted RemoteCacheManager. Use RemoteCacheManager.start before using the remote cache.";
+                throw new Infinispan.HotRod.Exceptions.RemoteCacheManagerNotStartedException(message);
+            }
+            else
+            {
+                return Task.Run(() => Put(key, val, lifespan, lifespanUnit, maxIdleTime, maxIdleUnit));
+            }
+        }
+
+        public async Task<V> PutIfAbsentAsync(K key, V val, ulong lifespan = 0, TimeUnit lifespanUnit = TimeUnit.SECONDS, ulong maxIdleTime = 0, TimeUnit maxIdleUnit = TimeUnit.SECONDS)
+        {
+            return await Task.Run(() => PutIfAbsent(key, val, lifespan, lifespanUnit, maxIdleTime, maxIdleUnit));
+        }
+
+        public Task<V> ReplaceAsync(K key, V val, ulong lifespan = 0, TimeUnit lifespanUnit = TimeUnit.SECONDS, ulong maxIdleTime = 0, TimeUnit maxIdleUnit = TimeUnit.SECONDS)
+        {
+            if (!manager.IsStarted())
+            {
+                String message = "Cannot perform operations on a cache associated with an unstarted RemoteCacheManager. Use RemoteCacheManager.start before using the remote cache.";
+                throw new Infinispan.HotRod.Exceptions.RemoteCacheManagerNotStartedException(message);
+            }
+            else
+            {
+                return Task.Run(() => Replace(key, val, lifespan, lifespanUnit, maxIdleTime, maxIdleUnit));
```

Comment:
If the method is marked with async it cannot raise exceptions and this leads to test failure (e.g. CacheManagerNotStartedTest.testPutAsync()). With the current implementation the Async methods just execute the operation in a parallell task and can't block so there nothing to await.